scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -271,14 +271,14 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
 
                 if (config.createMissingArchives) {
                     val packagePath = pkg.id.toPath()
-                    val missingArchives = HashSet<Provenance>()
 
-                    storedResults.mapNotNullTo(mutableSetOf()) { result ->
+                    val missingArchives = storedResults.mapNotNullTo(mutableSetOf()) { result ->
                         result.provenance.takeUnless { archiver.hasArchive("$packagePath/${it.hash()}") }
                     }
 
                     if (missingArchives.isNotEmpty()) {
                         val downloadResult = Downloader.download(pkg, downloadDirectory.resolve(packagePath))
+
```

Comment:
Nit: Unintended change?

Reply:
nope, it was intended. See the commit message ;-)

Reply:
Oops, sorry, as the first discussion post looked like a commit message, I thought there's only one commit in the PR!